### PR TITLE
ENH: show_versions added

### PR DESCRIPTION
**Pull request by thomaspinder**
Actioning Issue #899. Based upon code https://github.com/pandas-dev/pandas/blob/master/pandas/util/_print_versions.py.

### CODE REVIEW

File: geopandas/tools/_print_versions.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    return blob
+
+
+def show_versions(as_json=False):
+    """
+    Print system information and installed module versions.
+
+    Example
+    -------
+    > python -c "import geopandas as gpd; gpd.show_versions()"
```
Comment (martinfleis) [inline geopandas/tools/_print_versions.py]: Can you use `import geopandas` and then `geopandas.show_versions()` in the example for consistency of docs?

**Comment by jorisvandenbossche** [comment]
@thomaspinder indeed, as Martin said, very sorry about the slow follow-up here. And thanks for looking into this.

One general remark is that IMO the section on the system characteristics is too long for geopandas' purposes.

Eg comparing pandas vs sklearn:

```
In [27]: sklearn.show_versions()

System:
    python: 3.7.3 | packaged by conda-forge | (default, Mar 27 2019, 23:01:00)  [GCC 7.3.0]
executable: /home/joris/miniconda3/envs/dev37/bin/python
   machine: Linux-4.4.0-148-generic-x86_64-with-debian-stretch-sid

BLAS:
    macros: HAVE_CBLAS=None
  lib_dirs: /home/joris/miniconda3/envs/dev37/lib
cblas_libs: blas, cblas, lapack, blas, cblas, lapack, blas, cblas, lapack

Python deps:
       pip: 19.0.2
setuptools: 40.8.0
   sklearn: 0.21.dev0
     numpy: 1.16.2
     scipy: 1.2.1
    Cython: 0.29.5
    pandas: 0.25.0.dev0+590.g44d5498d8

In [28]: pd.show_versions() 

INSTALLED VERSIONS
------------------
commit: None
python: 3.7.3.final.0
python-bits: 64
OS: Linux
OS-release: 4.4.0-148-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.25.0.dev0+590.g44d5498d8
pytest: 4.4.0
pip: 19.0.2
setuptools: 40.8.0
Cython: 0.29.5
numpy: 1.16.2
scipy: 1.2.1
pyarrow: 0.13.0
xarray: None
IPython: 7.6.0.dev
sphinx: 1.8.4
patsy: 0.5.1
dateutil: 2.8.0
pytz: 2018.9
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 3.0.2
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml.etree: 4.3.2
bs4: 4.7.1
html5lib: None
sqlalchemy: 1.3.1
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: 0.3.1
pandas_gbq: None
pandas_datareader: None
gcsfs: None
```

I don't necessarily like the formatting of sklearn, but I do like the conciseness / compactness of the system information compared the pandas version (or something in between might be good as well). In general, I think we mainly want to know the OS platform, the python version and where this comes from (eg Anaconda or not). Also, the executable in sklearns version can be useful to see if somebody is using an environment. 
In general, I think for a start we should only add those items we think is informative for debugging purposes of typical geopandas reported issues. We can always add extra information later if we run into issues where we need more information.

**Comment by jorisvandenbossche** [comment]
As inspiration: sklearn's version is here: https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/utils/_show_versions.py

**Comment by martinfleis** [comment]
Hi @thomaspinder, are you willing to make some changes based on the discussion above or should someone else take over this PR? Thanks!

**Comment by martinfleis** [comment]
As @thomaspinder did not respond, I have re-done code in this PR based on sklearn.show-versions and included tests. Now it returns following:

    In [3]:  gpd.show_versions()

    SYSTEM INFO
    -----------
    python    : 3.7.3 | packaged by conda-forge | (default, Jul  1 2019, 14:38:56)  [Clang 4.0.1 (tags/RELEASE_401/final)]
    executable: /Users/martin/anaconda3/envs/guide/bin/python
    machine   : Darwin-18.6.0-x86_64-i386-64bit
    
    PYTHON DEPENDENCIES
    -------------------
    geopandas : 0.4.0+56.g9bd5ca9.dirty
    pandas    : 0.25.0
    fiona     : 1.8.6
    osgeo.gdal: 2.4.2
    numpy     : 1.17.0
    shapely   : 1.6.4.post2
    rtree     : 0.8.3
    pyproj    : 2.2.1
    Cython    : None
    matplotlib: 3.1.1

@jorisvandenbossche do you want to add any other info? I've decided not to change sklearn's formatting of the first part, just tweaked the rest.

**Comment by jorisvandenbossche** [comment]
That looks good!

We probably want some more optional packages (eg mapclassify, pysal). And cython should not be needed. 

Ideally, I would like to give more information for GEOS and GDAL (and now maybe also PROJ): which version is picked up (eg a system one or the conda installed one). But getting this information might be a bit more tricky, so certainly fine to keep that for later (this is certainly a good start to have).

**Comment by martinfleis** [comment]
@jorisvandenbossche I have added optional deps. I would skip GEOS, GDAL and PROJ for now, as it is indeed tricky. We can open an issue for that to keep that in mind for later.

**Comment by jorisvandenbossche** [comment]
Ah, this is giving some errors now ..

Apparently the older `mapclassify` has no `__version__` (the 1.0.1 one, used for Python 2.7). Maybe add a try/except around that one

**Comment by martinfleis** [comment]
Fixed. Merging.

**Comment by martinfleis** [comment]
Thank you @thomaspinder for starting this PR!

**Comment by jorisvandenbossche** [comment]
And thanks @martinfleis for the follow-up !

**Pull request merged**